app/models/soft_delete.py

```diff
@@ -0,0 +1,40 @@
+from django.contrib.auth.models import UserManager
+from django.db import models
+from django.utils import timezone
+
+
+class SoftDeleteManager(models.Manager):
+    def get_queryset(self):
+        return super().get_queryset().filter(deleted_at__isnull=True)
+
+
+class SoftDeleteModel(models.Model):
+
+    deleted_at = models.DateTimeField(null=True, blank=True)
+    objects = SoftDeleteManager()
+    all_objects = models.Manager()
+
+    def soft_delete(self):
+        self.deleted_at = timezone.now()
+        self.save()
+
+    def restore(self):
+        self.deleted_at = None
+        self.save()
+
+    class Meta:
+        abstract = True
+
+
+class SoftDeleteUserManager(UserManager):
+    def get_queryset(self):
```

Comment:
You don't need to override this method.

Reply:
@umairwaheed Then sir models.User.objects.all() would return soft_deleted instances too..In most cases we would need active users only..In case we need to get deleted users too, we can use models.User.all_objects.all() like we did while creating a new accounts.